ElectricCar.hpp

```diff
@@ -1,16 +1,22 @@
+#pragma once
+
+#include "Car.hpp"
 #include "ElectricEngine.hpp"
 
-class ElectricCar
-{
+class ElectricCar : virtual public Car {
 public:
     ElectricCar(ElectricEngine* engine);
-    ~ElectricCar();
-    void turnLeft();
-    void turnRight();
-    void brake();
-    void accelerate(int speed);
+
+    // Override from Car
+    ~ElectricCar() override;
+    void restore() override;
+    void changeEngine(Engine* engine) override;
```

Comment:
Isn't better to take specific type of engine for car. Like electric engine for electric car?

Reply:
What if it's a hybrid car?

Reply:
> Isn't better to take specific type of engine for car. Like electric engine for electric car?
What if it's a hybrid car?